kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -43,33 +62,56 @@ static String secretName(KafkaProxy primary) {
     @Override
     protected Secret desired(KafkaProxy primary,
                              Context<KafkaProxy> context) {
-
+        // formatter=off
         return new SecretBuilder()
                 .editOrNewMetadata()
-
-                .withName(secretName(primary))
-                .withNamespace(primary.getMetadata().getNamespace())
+                    .withName(secretName(primary))
+                    .withNamespace(primary.getMetadata().getNamespace())
                 .endMetadata()
-                .withStringData(Map.of(CONFIG_YAML_KEY,
-                        """
-                                adminHttp:
-                                  endpoints:
-                                    prometheus: {}
-                                virtualClusters:
-                                  demo:
-                                    targetCluster:
-                                      bootstrap_servers: TARGET_BOOTSTRAP_SERVERS
-                                    clusterNetworkAddressConfigProvider:
-                                      type: PortPerBrokerClusterNetworkAddressConfigProvider
-                                      config:
-                                        bootstrapAddress: localhost:9292
-                                        brokerAddressPattern: BROKER_ADDRESS_PATTERN
-                                    logNetwork: false
-                                    logFrames: false
-                                """
-                                .replace(
-                                        "TARGET_BOOTSTRAP_SERVERS", primary.getSpec().getBootstrapServers())
-                                .replace("BROKER_ADDRESS_PATTERN", ProxyService.serviceName(primary))))
+                .withStringData(Map.of(CONFIG_YAML_KEY, generateProxyConfig(primary)))
                 .build();
     }
+
+    String generateProxyConfig(KafkaProxy primary) {
+
+        List<Clusters> clusters = ClustersUtil.distinctClusters(primary);
+        var virtualClusters = clusters.stream()
+                .collect(Collectors.toMap(
+                        Clusters::getName,
+                        cluster -> getVirtualCluster(primary, cluster),
+                        (v1, v2) -> v1, // Dupes handled below
```

Comment:
How do we end up with dupe clusters here given we've gone through `distinctClusters` on line 77?